src/include/network/lan/address.rb

```diff
@@ -728,36 +704,96 @@ def HandleSlave(_key, event)
 
-      Ops.set(
-        @settings,
-        "SLAVES",
-        Convert.convert(
-          UI.QueryWidget(:msbox_items, :SelectedItems),
-          from: "any",
-          to:   "list <string>"
-        )
-      )
-      Ops.set(@settings, "BONDOPTION", UI.QueryWidget(Id("BONDOPTION"), :Value))
+      @settings["SLAVES"] = UI.QueryWidget(:msbox_items, :SelectedItems) || []
+
+      @settings["BONDOPTION"] = UI.QueryWidget(Id("BONDOPTION"), :Value).to_s
 
-      LanItems.bond_slaves = Ops.get_list(@settings, "SLAVES", [])
-      LanItems.bond_option = Ops.get_string(@settings, "BONDOPTION", "")
+      LanItems.bond_slaves = @settings["SLAVES"]
+      LanItems.bond_option = @settings["BONDOPTION"]
 
       # create list of "unconfigured" slaves
-      new_slaves = Builtins.filter(Ops.get_list(@settings, "SLAVES", [])) do |slave|
-        !Builtins.contains(configured_slaves, slave)
+      new_slaves = @settings["SLAVES"].select do |slave|
+        !configured_slaves.include? slave
       end
 
-      Lan.bond_autoconf_slaves = Convert.convert(
-        Builtins.toset(Builtins.merge(Lan.bond_autoconf_slaves, new_slaves)),
-        from: "list",
-        to:   "list <string>"
-      )
+      Lan.bond_autoconf_slaves = (Lan.bond_autoconf_slaves + new_slaves).uniq.sort
 
       nil
     end
 
+    # Validates created bonding. Currently just prevent the user to create a
+    # bond with more than one interface sharing the same physical port id
+    #
+    # @param [String] key the widget being validated
+    # @param [Hash] event the event being handled
+    # @return true if valid or user decision if not
+    def ValidateBond(_key, _event)
+      items = UI.QueryWidget(:msbox_items, :SelectedItems) || []
+      physical_port_ids = {}
+
+      items.each do |i|
+        if has_physical_port_id?(i)
+          p = physical_port_ids[physical_port_id(i)] ||= []
+          p << i
+        end
+      end
+
+      physical_port_ids.select! { |_k, v| v.size > 1 }
+
+      return true if physical_port_ids.empty?
+
+      message = ""
+
+      physical_port_ids.each do |port, devs|
+        label =  "PhysicalPortID (#{port}): "
+
+        message << wrap_text(devs.join(", "), 76, prepend_text: label)
+        message << "\n"
+      end
+
+      Popup.YesNoHeadline(
+        Label.WarningMsg,
+        _("The interfaces selected share the same physical port and bonding " \
+          "them \nmay not have the desired effect of redundancy.\n\n%s\n" \
+          "Really continue?\n") % message
+      )
+    end
+
+    # Wrap given text breaking lines longer than given wrap size. It supports
+    # cusom separator, max number of lines to split in and cut text to add
+    # as last line if cut was needed.
+    #
+    # @param [String>] text to be wrapped
+    # @param [String] wrap size
+    # @param [Hash <String>] optional parameters as separator and prepend_text.
+    # @return [String] wrap text
+    def wrap_text(text, wrap = 78, options = {})
```

Comment:
About named arguments.. for me it's about taste of code also. In this case are very strange parameters, and of course you know which parameters are available without reading method implementation but for this options i don't like named arguments at all.